### PR TITLE
Fix multiple flakiness in integration tests

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Fix flakiness in multiple tests and skips some on Windows for now. 

## Issue reference

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by daixiang0 — APPROVED** [review]
LGTM

**Comment by artursouza** [comment]
> Add time wait is common way to fix flakiness, LGTM

I am also fixing some weird flakiness in connecting to localhost, using 127.0.0.1 since we have seen GH actions resolving localhost to different ip addresses.

**Comment by daixiang0** [comment]
> > Add time wait is common way to fix flakiness, LGTM
> 
> I am also fixing some weird flakiness in connecting to localhost, using 127.0.0.1 since we have seen GH actions resolving localhost to different ip addresses.

Good catch! Maybe we can send the issue to GH community.

File: tests/integration/framework/process/grpc/operator/operator.go [hunk 1]
```diff
@@ -156,8 +156,8 @@ func New(t *testing.T, fopts ...Option) *Operator {
 	o.GRPC = procgrpc.New(t, append(opts.grpcopts,
 		procgrpc.WithServerOption(func(t *testing.T, ctx context.Context) grpc.ServerOption {
 			secProv, err := security.New(ctx, security.Options{
-				SentryAddress:           "localhost:" + strconv.Itoa(opts.sentry.Port()),
-				ControlPlaneTrustDomain: "localhost",
+				SentryAddress:           "127.0.0.1:" + strconv.Itoa(opts.sentry.Port()),
```
Comment (ItalyPaleAle) [inline tests/integration/framework/process/grpc/operator/operator.go:159]: This will break on IPv6-only environments

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

File: tests/integration/framework/process/grpc/operator/operator.go (hunk 1 above)
Comment (artursouza) [reply tests/integration/framework/process/grpc/operator/operator.go]: For testing, it is OK if this fixes it. If not, I revert.

**Comment by artursouza** [comment]
Linux and MacOS int tests seem to be more stable now. On Windows, the subset of failing tests is the same and consistently - I recommend investigating locally on a Windows machine as I am not sure it is just a test issue anymore because nothing made an impact on those tests on Windows.

File: tests/integration/framework/client/etcd.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+// Returns an adapted Etcd client for tests.
+func Etcd(t *testing.T, cfg clientv3.Config) *EtcdClient {
+	etcdClient, err := clientv3.New(cfg)
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, etcdClient.Close()) })
+
+	return &EtcdClient{etcdClient}
+}
+
+func (c *EtcdClient) ListAllKeys(ctx context.Context, prefix string) ([]string, error) {
```
Comment (artursouza) [inline tests/integration/framework/client/etcd.go:37]: This is a handy helper method to make sure the list of keys on Etcd consider pagination and iterates on them.

**Review by yaron2 — APPROVED** [review]

**Pull request merged**